Demo/MozillaSocial-iOS/MoSoContent/Auth.swift

```diff
@@ -0,0 +1,138 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import AuthenticationServices
+
+let isProd = false
+let baseURL = isProd ? "https://mozilla.social" : "https://stage.moztodon.nonprod.webservices.mozgcp.net"
+let redirectScheme = "mozillasocial"
+let redirectURI = redirectScheme + "://auth"
+
+class Auth: NSObject, ASWebAuthenticationPresentationContextProviding {
+
+    var authToken: Token?
+
+    func presentationAnchor(for session: ASWebAuthenticationSession) -> ASPresentationAnchor {
+        return ASPresentationAnchor()
+    }
+
+    func registerApp() async -> ClientEntity? {
```

Comment:
any chance this could become a throwing method, and we catch the error at the caller site? Probably doesn't matter much in this scenario, but Ideally a caller would know the error and decide how to handle it

Reply:
Given that this code is almost solely to unblock Recommendations and will be replaced by a much more fleshed out version soon, I only considered the happy path.
I do very much agree that functions like this should throw errors for it's caller to handle, but in this particular case I think we can skip it.